CalibCalorimetry/HcalAlgos/src/HcalDbHardcode.cc

```diff
@@ -564,6 +564,28 @@ HcalQIECoder HcalDbHardcode::makeQIECoder (HcalGenericDetId fId) {
   return result;
 }
 
+HcalQIEType HcalDbHardcode::makeQIEType (HcalGenericDetId fId, bool testHFQIE10) {
+  int qieType = HcalQIEType::QIE8; //default
+  if (testHFQIE10){ //2016 test
+    if(fId.genericSubdet() == HcalGenericDetId::HcalGenForward && fId.isHcalDetId()) {
+        HcalDetId hid(fId);
+        if(hid.depth()>=3) qieType = HcalQIEType::QIE10;
+    }
```

Comment:
Hello @kpedro88 

we understand that testHFQIE10 serves the purpose of switching at python level to the 2016 scenario (if set to true), where only a few channels need be handled with the QIE10.
If set to false, the HCAL is seen as >=2017, after the full transition to the new readout.

Why is depth >=3 used as discriminant to set QIE10 ?
According to https://indico.cern.ch/event/492087/contribution/4/attachments/1225068/1792949/Phase2-Talk30.pdf  slide 4, only HB can have 3 levels of segmentation, not HF.

Furthermore, and more critically:
who is data <=2015 going to be handled ? We understand that this PR is not the final word - given than no DB content is actually used, no history influences how the HcalQIEType  are set up; it would teherefor appear that when running on data from the past, the HcalQIEType will be incorrectly set to QIE10.
Can you explain, pls ?

@ggovi @mmusich